cartographer/mapping/map_builder.cc

```diff
@@ -285,8 +285,12 @@ std::map<int, int> MapBuilder::LoadState(
                                  true);
   }
 
-  MapById<SubmapId, mapping::proto::Submap> submap_id_to_submap;
-  MapById<NodeId, mapping::proto::Node> node_id_to_node;
+  CHECK_NE(deserializer.header().format_version(),
+           io::kFormatVersionWithoutSubmapHistograms)
+      << "The pbstream file contains submaps without rotational histograms. "
+         "This can be converted, but the conversion was removed here due to "
+         "memory concerns.";
```

Comment:
This check message would need some changes probably.

Reply:
Might be best to mention the migration tool here. Also: could we change this to only fail for 3D data? As far as I understand nothing has changed for 2D, so old maps should just continue to work.

Reply:
done